CONTRIBUTING.md

```diff
@@ -134,67 +134,55 @@ Enhancement suggestions are tracked as [GitHub issues](https://github.com/Start9
-project back to them and have them hosted more visibly. The way we bootstrap through this problem is by downloading version
-[2.1.3](https://github.com/commercialhaskell/stack/releases/download/v2.1.3/stack-2.1.3-linux-arm.tar.gz) and using that
-to compile v2.5.1. Before you can successfully compile anything with GHC on the Raspberry Pi. You will need to tweak the
-relevant GHC config. You will need to edit the file at `~/.stack/programs/arm-linux/ghc-8.10.2/lib/ghc-8.10.2/settings`
-and change the line `("C compiler flags", " -marm -fno-stack-protector -mcpu=cortex-a7")` to include `-mcpu=cortex-a7`.
-You will also need to make sure you've downloaded and installed LLVM 9.
-
-Once you have done these things, you simply need to `cd` into the embassy-os project and then run `make agent`.
+
+First, clone the EmbassyOS repository and from the project root, pull in the submodules for dependent libraries.
+
+```
+git clone https://github.com/Start9Labs/embassy-os.git
+git submodule update --init --recursive
+```
 
 ##### ui
 - Requirements
   - [Install nodejs](https://nodejs.org/en/)
   - [Install npm](https://www.npmjs.com/get-npm)
   - [Install ionic cli](https://ionicframework.com/docs/intro/cli)
-- Scripts (run within ./ui directory)
-  - `npm i` installs ui node package dependencies
+- Installation: follow the setup instructions [here](ui/README.md)
+- Scripts (run within `./ui` directory)
+  - `npm --prefix ui install` installs ui node package dependencies
   - `npm run build` compiles project, depositing build artifacts into ./ui/www
-  - `npm run build-prod` as above but customized for deployment to an Embassy
+  - `npm run build-deps` compiles dependent libraries, particularly the client for patch-db.
+  - `npm --prefix ui run build-prod` as above but customized for deployment to an Embassy
   - `ionic serve` serves the ui on localhost:8100 for local development. Edit ./ui/use-mocks.json to 'true' to use mocks during local development
-  - `./build-send.sh <embassy .local address suffix>` builds the project and deploys it to the referenced Embassy
-    - Find your Embassy on the LAN using the Start9 Setup App or network tools. It's address will be of the form `start9-<suffix>.local`.
-    - For example `./build-send.sh abcdefgh` will deploy the ui to the Embassy with LAN address `start9-abcdefgh.local`.
-    - SSH keys must be installed on the Embassy prior to running this script.
 
-##### appmgr
+##### backend
   - [Install Rust](https://rustup.rs)
   - Recommended: [rust-analyzer](https://rust-analyzer.github.io/)
+  - [Docker](https://docs.docker.com/get-docker/)
+  - [Rust ARM64 Build Container](https://github.com/Start9Labs/rust-arm-builder)
+  - Scripts (run withing the `./backend` directory)
+    - `build-prod.sh` - compiles a release build of the artifacts for running on ARM64
+    - `build-dev.sh` - compiles a development build of the artifacts for running on ARM64
 
 #### Building The Image

```

Comment:
So we have ui, backend, and building the image. Should these all really be under "your first code contribution"? For example, contributing to the ui does not require the backend or building the image. Same for diagnositc UI and setup-wizard. This just feels a bit disorganized.